### ISSUE TITLE
InnerFunctionsSniff should flag functions defined inside closures

### ISSUE THREAD

**Issue by Daimona** [squizlabs/PHP_CodeSniffer#3806]
InnerFunctionsSniff detects functions defined inside other functions, but not inside closures, meaning it won't report any violations for the following snippet:

```php
$myFunc = function () {
	function test(): int {
		return 54;
	}
};
```

For consistency, this should be treated the same as:

```php
function myFunc() {
	function test(): int {
		return 54;
	}
}
```

**Comment by jrfnl** [comment]
FYI: the fix for this issue is included in today's [PHP_CodeSniffer 3.8.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.8.0).

As per #3932, development on PHP_CodeSniffer will continue in the [PHPCSStandards/PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer) repository. If you want to stay informed, you may want to start "watching" that repo (or watching releases from that repo).
